mock/secrets_manager_client.go

```diff
@@ -3,46 +3,186 @@ package mock
 import (
 	"context"
 	"errors"
+	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/evergreen-ci/utility"
 )
 
+// StoredSecret is a representation of a secret kept in the global secret
+// storage cache.
+type StoredSecret struct {
+	Value       string
+	BinaryValue []byte
+	Created     time.Time
+}
+
+// GlobalSecretCache is a global secret storage cache that provides a simplified
+// in-memory implementation of a secrets storage service. This can be used with
+// the SecretsManagerClient to access and modify secrets, or used directly.
+var GlobalSecretCache map[string]StoredSecret
+
+func init() {
+	GlobalSecretCache = map[string]StoredSecret{}
+}
+
 // SecretsManagerClient provides a mock implementation of a
 // cocoa.SecretsManagerClient. This makes it possible to introspect on inputs
 // to the client and control the client's output. It provides some default
 // implementations where possible.
-type SecretsManagerClient struct{}
+type SecretsManagerClient struct {
+	CreateSecretInput  *secretsmanager.CreateSecretInput
+	CreateSecretOutput *secretsmanager.CreateSecretOutput
+
+	GetSecretValueInput  *secretsmanager.GetSecretValueInput
+	GetSecretValueOutput *secretsmanager.GetSecretValueOutput
+
+	UpdateSecretInput  *secretsmanager.UpdateSecretInput
+	UpdateSecretOutput *secretsmanager.UpdateSecretOutput
+
+	DeleteSecretInput  *secretsmanager.DeleteSecretInput
+	DeleteSecretOutput *secretsmanager.DeleteSecretOutput
+}
 
 // CreateSecret saves the input options and returns a new mock secret. The mock
-// output can be customized. By default, it will create a cached mock secret
-// based on the input.
+// output can be customized. By default, it will create and save a cached mock
+// secret based on the input in the global secret cache.
 func (c *SecretsManagerClient) CreateSecret(ctx context.Context, in *secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error) {
-	return nil, errors.New("TODO: implement")
+	c.CreateSecretInput = in
+
+	if c.CreateSecretOutput != nil {
+		return c.CreateSecretOutput, nil
+	}
+
+	if in.Name == nil {
+		return nil, errors.New("missing secret name")
+	}
+	if in.SecretBinary != nil && in.SecretString != nil {
+		return nil, errors.New("cannot specify both secret binary and secret string")
+	}
+	if in.SecretBinary == nil && in.SecretString == nil {
+		return nil, errors.New("must specify either secret binary or secret string")
+	}
+
+	if in.SecretBinary != nil {
```

Comment:
Nit: since you check that both `in.SecretBinary` and `in.SecretString` are not specified, you can omit the if statements and just do:
```
GlobalSecretCache[*in.Name] = StoredSecret{
			BinaryValue: in.SecretBinary,
                         Value: utlilty.FromStringPtr(in.SecretString), 
			Created:     time.Now(),
}
```

Reply:
Done.